// jest-cucumber/dist/specs/test-data/running-a-scenario.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.runningAScenarioStepsWithMissingStep = exports.runningAScenarioSteps = exports.runningAScenarioFeature = void 0;
exports.runningAScenarioFeature = "\nFeature: Test\n\nScenario: Doing some stuff\n    Given I did some stuff\n    When I do some stuff\n    Then I should have done some stuff\n";
exports.runningAScenarioSteps = function (mockFeature, defineMockFeature) {
    defineMockFeature(mockFeature, function (test) {
        test('Doing some stuff', function (_a) {
            var given = _a.given, when = _a.when, then = _a.then;
            given('I did some stuff', function () {
                // Nothing to do here
            });
            when('I do some stuff', function () {
                // Nothing to do here
            });
            then('I should have done some stuff', function () {
                // Nothing to do here
            });
        });
    });
};
exports.runningAScenarioStepsWithMissingStep = function (mockFeature, defineMockFeature) {
    defineMockFeature(mockFeature, function (test) {
        test('Doing some stuff', function (_a) {
            var given = _a.given, when = _a.when, then = _a.then;
            given('I did some stuff', function () { });
            when('I do some stuff', function () { });
        });
    });
};
